food-waste-ui/src/components/App/App.jsx

```diff
@@ -119,34 +119,43 @@ export function MainApp({
           setCurrentUser(curUser);
 
           let allProducts = response.data.products;
-          //TODO: filter products for search
-          setProducts(allProducts);
+          let newProducts = allProducts.filter((item) =>
+            item.name.toLowerCase().includes(searchTerm.toLowerCase())
+          );
+          setProducts(newProducts);
           const userProducts = allProducts.filter(
             (item) => item.user.objectId == curUser.objectId
           ); //the thing with setState it is async, if setting state and doing based off state after might not update
           setMyProducts(userProducts);
-          setIsLoading(false);
         });
       })
       .catch((err) => {
         console.log(err);
```

Comment:
When you have not implemented something yet you can add a comment that looks like
// TODO @cwang12197 : Add error logging and error screen